src/main/java/iudx/catalogue/server/database/DatabaseServiceImpl.java

```diff
@@ -247,17 +248,21 @@ public DatabaseService createItem(JsonObject doc, Handler<AsyncResult<JsonObject
               });
             } else {
               /* Insert document */
-              client.docPostAsync(doc.toString(), postRes -> {
-                if (postRes.succeeded()) {
-                  handler.handle(Future.succeededFuture(
-                        respBuilder.withStatus(SUCCESS)
-                        .withResult(id, INSERT, SUCCESS)
-                        .getJsonResponse()));
-                } else {
-                  handler.handle(Future.failedFuture(errorJson));
-                  LOGGER.error("Fail: Insertion failed" + postRes.cause());
+              new Timer().schedule(new TimerTask() {
```

Comment:
Is this function calling elastic search after 3 seconds or giving back the response after 3 seconds?

Reply:
Its the delay before the task to be executed. So calling elastic search after 3 seconds.